### PR TITLE
added a bearer token verify API

### PR DESCRIPTION
**Pull request by woutslakhorst**
This API can be used by a reverse proxy to do a delegated auth check.

For example, nginx: https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-subrequest-authentication/

This will help in testing and endpoints that have no care context.

### CODE REVIEW

File: docs/_static/nuts-auth.yaml [hunk 1]
```diff
@@ -160,6 +160,33 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/AccessTokenRequestFailedResponse"
+  /auth/accesstoken/verify:
+    head:
+      operationId: verifyAccessToken
+      summary: |
+        Verifies the access token given in the Authorization header (as bearer token). If it's a valid access token issued by this server, it'll return a 200 status code.
```
Comment (reinkrul) [inline docs/_static/nuts-auth.yaml:167]: Ah, so this endpoint does not intent to implement RFC7662 (https://tools.ietf.org/html/rfc7662) but is sort of a helper API for proxies. I think that's a good solution vs. implementing the RFC in a disputed way.

File: docs/_static/nuts-auth.yaml [hunk 2]
```diff
@@ -160,6 +160,33 @@ paths:
...
+        '200':
+          description: The access token is valid. It has been signed by this server.
+          content:
+            text/plain:
+              schema:
+                type: string
+        '403':
+          description: The given access token is invalid.
+          content:
+            text/plain:
```
Comment (reinkrul) [inline docs/_static/nuts-auth.yaml:187]: the implementation returns no content

File: docs/_static/nuts-auth.yaml [hunk 3]
```diff
@@ -160,6 +160,33 @@ paths:
...
+            type: string
+      responses:
+        '200':
+          description: The access token is valid. It has been signed by this server.
+          content:
+            text/plain:
+              schema:
+                type: string
+        '403':
+          description: The given access token is invalid.
```
Comment (reinkrul) [inline docs/_static/nuts-auth.yaml:185]: or it couldn't be verified

File: api/api.go [hunk 4]
```diff
@@ -325,3 +326,30 @@ func (api *Wrapper) IntrospectAccessToken(ctx echo.Context) error {
 
 	return ctx.JSON(http.StatusOK, introspectionResponse)
 }
+
+const bearerPrefix = "bearer "
+
+// VerifyAccessToken verifies if a request contains a valid bearer token issued by this server
+func (api *Wrapper) VerifyAccessToken(ctx echo.Context, params VerifyAccessTokenParams) error {
+	if len(params.Authorization) == 0 {
+		logrus.Warn("no authorization header given")
```
Comment (reinkrul) [inline api/api.go:335]: We usually start log messages with a capital (first character)

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: docs/_static/nuts-auth.yaml (hunk 2 above)
Comment (woutslakhorst) [reply docs/_static/nuts-auth.yaml:187]: removed from spec

**Review by reinkrul — APPROVED** [review]

**Pull request merged**